### PR TITLE
sql: Standardize unavailable timezone error prefix

### LINKED ISSUE #3925: TestLogic: improper/incomplete test for timezone errors
The `datetime` test in TestLogic expects an incorrect/invalid timezone to fail with "cannot find ..." from Go's time library. However due to the way Go's time library is written, combined with how the distrib's standard package installs Go on (at least) Ubuntu 15, this error cannot be produced upon selecting an invalid timezone; instead the error shows up as "unable to find zoneinfo.zip", and thus:

```
testdata/datetime:309: SET TIME ZONE 'foobar'
--- FAIL: TestLogic (1.89s)
        logic_test.go:459: testdata/datetime:309: expected "cannot find foobar.*", but found
                pq: open /usr/lib/go/lib/time/zoneinfo.zip: no such file or directory
```

The problem is that `zoneinfo.zip` is not necessary on Unix (and thus is not installed by Go's package on Ubuntu 15), yet time's lookup code will fall back to it if it cannot find the selected timezone in the standard locations. 

The proper fix / workaround is to accept an unavailable zoneinfo.zip as a correct error message for an invalid timezone.

### PR DESCRIPTION
**Pull request by nvb**
Closes #3925.

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/3960)

### CODE REVIEW

File: sql/set.go [hunk 1]
```diff
@@ -94,7 +94,7 @@ func (p *planner) SetTimeZone(n *parser.SetTimeZone) (planNode, *roachpb.Error)
 			location = "UTC"
 		}
 		if _, err := time.LoadLocation(location); err != nil {
-			return nil, roachpb.NewError(err)
+			return nil, roachpb.NewUErrorf("cannot find %s: %v", location, err)
```
Comment (tamird) [inline sql/set.go]: %s for error. should we use this opportunity to give a better error message? pg gives `ERROR:  invalid value for parameter "TimeZone": "blabla"`

Comment (nvanbenschoten) [inline sql/set.go]: Are you sure about %s for error? I've seen it both ways, so I double checked before pushing this and it seems like Docker uses %v for errors. So does http://blog.golang.org/error-handling-and-go.

Comment (bdarnell) [inline sql/set.go]: If `err` is known to be non-nil, `%s` and `%v` are equivalent. If `err` may be nil, `%v` is a little nicer (https://play.golang.org/p/6g1NUFTI3r). I prefer `%v` for this reason but `%s` is currently more common in cockroach code.

Comment (bdarnell) [inline sql/set.go]: Also the requested location name should use `%q` so it gets quoted: `"cannot find time zone %q: %v"`

Comment (nvanbenschoten) [inline sql/set.go]: Done.

**Comment by tamird** [comment]
lgtm

**Pull request merged**